src/ctap/command.rs

```diff
@@ -134,19 +138,21 @@ impl TryFrom<cbor::Value> for AuthenticatorMakeCredentialParameters {
         let cred_param_vec = read_array(ok_or_missing(param_map.get(&cbor_unsigned!(4)))?)?;
         let mut pub_key_cred_params = vec![];
         for cred_param_map_value in cred_param_vec {
-            let cred_param_map = read_map(cred_param_map_value)?;
-            let cred_type = PublicKeyCredentialType::try_from(ok_or_missing(
-                cred_param_map.get(&cbor_text!("type")),
-            )?)?;
-            let alg = read_integer(ok_or_missing(cred_param_map.get(&cbor_text!("alg")))?)?;
-            pub_key_cred_params.push((cred_type, alg));
+            if let Ok(cred_param) = PublicKeyCredentialParameter::try_from(cred_param_map_value) {
+                pub_key_cred_params.push(cred_param);
+            }
         }
 
         let exclude_list = match param_map.get(&cbor_unsigned!(5)) {
             Some(entry) => {
                 let exclude_list_vec = read_array(entry)?;
                 let mut exclude_list = vec![];
```

Comment:
Nit: this vector should `reserve()` some space here, given that the length is known (`let count = std::cmp::min(exclude_list_vec.len(), MAX_CREDENTIAL_COUNT_IN_LIST.unwrap_or(usize::MAX))`).

Reply:
For now we only accept 1 algorithm in the current implementation, so the vector has length 0 or 1 for specification-conform calls. Shouldn't hurt, given that we need the count for take anyway.

Reply:
Update: I changed it to a collect construction, that solves the problem more elegantly.